hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/expression/json/JsonValueFunction.java

```diff
@@ -218,13 +229,23 @@ public void writeData(final ObjectDataOutput out) throws IOException {
         super.writeData(out);
         out.writeString(onEmpty.name());
         out.writeString(onError.name());
+        out.writeObject(constantPathCache);
     }
 
     @Override
     public void readData(final ObjectDataInput in) throws IOException {
         super.readData(in);
         this.onEmpty = SqlJsonValueEmptyOrErrorBehavior.valueOf(in.readString());
         this.onError = SqlJsonValueEmptyOrErrorBehavior.valueOf(in.readString());
+        this.constantPathCache = in.readObject();
+        if (this.constantPathCache == null) {
+            this.pathCache = JsonPathUtil.makePathCache();
+        }
+    }
+
+    private void readObject(ObjectInputStream stream) throws ClassNotFoundException, IOException {
+        stream.defaultReadObject();
+        prepareCache();
```

Comment:
`prepareCache()` compiles the constant operand into `constantPathCache`. But as we benchmarked, it's better to serde it rather than recompile it, same as we do when IDS is used. We need to make `constantPathCache` non-transient and initialize `pathCache` here, if `constantPathCache` is null after deser.

Reply:
Done